### ISSUE TITLE
tf.keras.models.load_model outputs random evaluation values

### ISSUE THREAD

**Issue by hamidkhb** [keras-team/keras#14240]
**Important note:** the development of Keras is currently taking place at [`github.com/tensorflow/tensorflow`](https://github.com/tensorflow/tensorflow). Please file your bug report by creating a new issue in the TensorFlow repository.

Please note that multi-backend Keras development has been discontinued. Do not report issues about multi-backend Keras (Keras 2.3.1 and lower), only report issues about the TensorFlow implementation of Keras (tf.keras).

I'm using Tensorflow 2.3 on GPU: nvidia Tesla v100. After loading a model the evaluation outputs random values:

```
model.evaluate(source_gen)
model.evaluate(target_gen)
```

> 2984/2984 [==============================] - 5s 2ms/step - loss: 0.1515 - accuracy: 0.9811
>4634/4634 [==============================] - 8s 2ms/step - loss: 1.4460 - accuracy: 0.7262

```
model.save_weights("models/weights.h5")
model.save("models/test.h5")

model_2 = tf.keras.models.load_model("models/test.h5")
model_2.load_weights("models/weights.h5")
model_2.evaluate(source_gen)
model_2.evaluate(target_gen)
```
>2984/2984 [==============================] - 5s 2ms/step - loss: 0.1515 - accuracy: 0.2070
>4634/4634 [==============================] - 8s 2ms/step - loss: 1.4460 - accuracy: 0.2473

also tried:

```
model.save("models/test.h5")
model_2 = tf.keras.models.load_model("models/test.h5")
model_2.evaluate(source_gen)
model_2.evaluate(target_gen)
```
>2984/2984 [==============================] - 5s 2ms/step - loss: 0.1515 - accuracy: 0.2070
>4634/4634 [==============================] - 8s 2ms/step - loss: 1.4460 - accuracy: 0.2473

**Comment by jvishnuvardhan** [comment]
@hamidkhb I think this is not an issue with saving and loading model or model weights. I tried to train, evaluate, save and then load as `model_2`, and finally evaluated but the results are same from `model` and `model_2`. Please check the [gist here](https://colab.research.google.com/gist/jvishnuvardhan/77e9206d6804b97961a2d5db4a4ae347/untitled1000.ipynb).

I think the root-cause is you are using data generators which generates different set of data for `model` and `model_2` which  results in different results from the evaluations. Thanks!

**Comment by jvishnuvardhan** [comment]
Closing due to lack of recent activity. Please update the issue when new information becomes available, and we will reopen the issue. Thanks!
